dashboard/client/src/common/ProfilingLink.component.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import React from "react";
+import "@testing-library/jest-dom";
+import { ProfilerButton } from "./ProfilingLink";
+
+describe("ProfilerButton", () => {
+  const mockProps = {
+    profilerUrl: "http://localhost:3000/worker/memory_profile",
+    type: "",
+  };
+  it("renders button correctly", () => {
+    render(<ProfilerButton {...mockProps} />);
+    const button = screen.getByLabelText(/Memory Profiling/);
+    expect(button).toBeInTheDocument();
+  });
+
+  it("opens the dialog when the button is clicked", async () => {
+    render(<ProfilerButton {...mockProps} />);
+    const button = screen.getByLabelText(/Memory Profiling/);
+
+    fireEvent.click(button);
```

Comment:
Can you use userEvent instead of fireEvent? You can see examples in other parts of the code